llvm/lib/LTO/ThinLTOCodeGenerator.cpp

```diff
@@ -89,6 +94,32 @@ namespace {
 // thred per core, as indicated by the usage of
 // heavyweight_hardware_concurrency() below.
 static cl::opt<int> ThreadCount("threads", cl::init(0));
+static cl::opt<bool> CacheLogging(
+    "thinlto-cache-logging", cl::desc("Enable logging for thinLTO caching"),
+    cl::init((bool)sys::Process::GetEnv("LLVM_THINLTO_CACHE_LOGGING")),
```

Comment:
Can we avoid getenv here in the C++ implementation and move it up to liblto and/or whatever executable calls this? Same for the other one below. This pattern hides impure checks deep in the library code.

Reply:
I don't see this make any difference. Also the environment checks doesn't affect the output of the computation.

Reply:
If we capture stderr as part of the output of the linker like we do in clang, this affects the output since you're logging to stderr.  It looks like the experimental lld caching does capture stderr, for example. I know that doesn't apply to thinlto caching that you are adding here, but it seems like a bad idea to have getenv calls here.  

I agree the determinism check is technically fine, though personally I would prefer we have no getenv at all in the underlying implementation.

Reply:
LLD cannot hit this code path. Everything in this file is libLTO only. The logging is for service, we do not replay the log. This is the same as the remarks for cache hit/miss, it is going to be different for replay.